src/H5T.c

```diff
@@ -3489,18 +3493,24 @@ H5T__initiate_copy(const H5T_t *old_dt)
     /* Copy shared information */
     *(new_dt->shared) = *(old_dt->shared);
 
-    /* Reset VOL fields */
+    /* Increment ref count on owned VOL object */
+    if (new_dt->shared->owned_vol_obj)
+        new_dt->shared->owned_vol_obj->rc++;
```

Comment:
Having other parts of the library modify fields in the H5VL structs is going to be difficult to maintain.   Could you please add a "increment refcount" routine in the H5VL code and call that?

Reply:
I can do this, though looking through the code there are many other places that directly manipulate the H5VL_object_t struct outside the H5VL package

Reply:
No doubt!   I would suggest that you just herd as many cats as you have time for now and we'll keep tracking them down over time.